### DISCUSSION TITLE
git Upgrade from v3.3.9 to v3.4.1 failed with PostgreSQL Error

### DISCUSSION THREAD

**Discussion by Abulhallaj** [netbox-community/netbox#11350] [Q&A]
git Upgrade from v3.3.9 to v4.4.1 failed 
current postgre is 10.22 
any idea?

`Using Python 3.9.16
Removing old virtual environment...
Creating a new virtual environment at /opt/netbox/venv...
Updating pip (pip install --upgrade pip)...
Requirement already satisfied: pip in ./venv/lib/python3.9/site-packages (22.0.4)
Collecting pip
  Using cached pip-22.3.1-py3-none-any.whl (2.1 MB)
Installing collected packages: pip
  Attempting uninstall: pip
    Found existing installation: pip 22.0.4
    Uninstalling pip-22.0.4:
      Successfully uninstalled pip-22.0.4
Successfully installed pip-22.3.1
pip 22.3.1 from /opt/netbox/venv/lib/python3.9/site-packages/pip (python 3.9)
Installing Python system packages (pip install wheel)...
Collecting wheel
  Using cached wheel-0.38.4-py3-none-any.whl (36 kB)
Installing collected packages: wheel
Successfully installed wheel-0.38.4
Installing core dependencies (pip install -r requirements.txt)...
Collecting bleach==5.0.1
  Using cached bleach-5.0.1-py3-none-any.whl (160 kB)
Collecting Django==4.1.4
  Using cached Django-4.1.4-py3-none-any.whl (8.1 MB)
Collecting django-cors-headers==3.13.0
  Using cached django_cors_headers-3.13.0-py3-none-any.whl (13 kB)
Collecting django-debug-toolbar==3.8.1
  Using cached django_debug_toolbar-3.8.1-py3-none-any.whl (221 kB)
Collecting django-filter==22.1
  Using cached django_filter-22.1-py3-none-any.whl (80 kB)
Collecting django-graphiql-debug-toolbar==0.2.0
  Using cached django_graphiql_debug_toolbar-0.2.0-py3-none-any.whl (7.3 kB)
Collecting django-mptt==0.14
  Using cached django_mptt-0.14.0-py3-none-any.whl (115 kB)
Collecting django-pglocks==1.0.4
  Using cached django_pglocks-1.0.4-py3-none-any.whl
Collecting django-prometheus==2.2.0
  Using cached django_prometheus-2.2.0-py2.py3-none-any.whl (29 kB)
Collecting django-redis==5.2.0
  Using cached django_redis-5.2.0-py3-none-any.whl (30 kB)
Collecting django-rich==1.4.0
  Using cached django_rich-1.4.0-py3-none-any.whl (7.0 kB)
Collecting django-rq==2.6.0
  Using cached django_rq-2.6.0-py2.py3-none-any.whl (48 kB)
Collecting django-tables2==2.4.1
  Using cached django_tables2-2.4.1-py2.py3-none-any.whl (93 kB)
Collecting django-taggit==3.1.0
  Using cached django_taggit-3.1.0-py3-none-any.whl (60 kB)
Collecting django-timezone-field==5.0
  Using cached django_timezone_field-5.0-py3-none-any.whl (10 kB)
Collecting djangorestframework==3.14.0
  Using cached djangorestframework-3.14.0-py3-none-any.whl (1.1 MB)
Collecting drf-yasg[validation]==1.21.4
  Using cached drf_yasg-1.21.4-py3-none-any.whl (6.4 MB)
Collecting graphene-django==3.0.0
  Using cached graphene_django-3.0.0-py2.py3-none-any.whl (101 kB)
Collecting gunicorn==20.1.0
  Using cached gunicorn-20.1.0-py3-none-any.whl (79 kB)
Collecting Jinja2==3.1.2
  Using cached Jinja2-3.1.2-py3-none-any.whl (133 kB)
Collecting Markdown==3.3.7
  Using cached Markdown-3.3.7-py3-none-any.whl (97 kB)
Collecting mkdocs-material==8.5.11
  Using cached mkdocs_material-8.5.11-py3-none-any.whl (7.6 MB)
Collecting mkdocstrings[python-legacy]==0.19.1
  Using cached mkdocstrings-0.19.1-py3-none-any.whl (25 kB)
Collecting netaddr==0.8.0
  Using cached netaddr-0.8.0-py2.py3-none-any.whl (1.9 MB)
Collecting Pillow==9.3.0
  Using cached Pillow-9.3.0-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (3.2 MB)
Collecting psycopg2-binary==2.9.5
  Using cached psycopg2_binary-2.9.5-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (3.0 MB)
Collecting PyYAML==6.0
  Using cached PyYAML-6.0-cp39-cp39-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_12_x86_64.manylinux2010_x86_64.whl (661 kB)
Collecting sentry-sdk==1.11.1
  Using cached sentry_sdk-1.11.1-py2.py3-none-any.whl (168 kB)
Collecting social-auth-app-django==5.0.0
  Using cached social_auth_app_django-5.0.0-py3-none-any.whl (24 kB)
Collecting social-auth-core[openidconnect]==4.3.0
  Using cached social_auth_core-4.3.0-py3-none-any.whl (343 kB)
Collecting svgwrite==1.4.3
  Using cached svgwrite-1.4.3-py3-none-any.whl (67 kB)
Collecting tablib==3.3.0
  Using cached tablib-3.3.0-py3-none-any.whl (48 kB)
Collecting tzdata==2022.7
  Using cached tzdata-2022.7-py2.py3-none-any.whl (340 kB)
Collecting jsonschema==3.2.0
  Using cached jsonschema-3.2.0-py2.py3-none-any.whl (56 kB)
Collecting webencodings
  Using cached webencodings-0.5.1-py2.py3-none-any.whl (11 kB)
Collecting six>=1.9.0
  Using cached six-1.16.0-py2.py3-none-any.whl (11 kB)
Collecting asgiref<4,>=3.5.2
  Using cached asgiref-3.6.0-py3-none-any.whl (23 kB)
Collecting sqlparse>=0.2.2
  Using cached sqlparse-0.4.3-py3-none-any.whl (42 kB)
Collecting django-js-asset
  Using cached django_js_asset-2.0.0-py3-none-any.whl (4.9 kB)
Collecting prometheus-client>=0.7
  Using cached prometheus_client-0.15.0-py3-none-any.whl (60 kB)
Collecting redis!=4.0.0,!=4.0.1,>=3
  Using cached redis-4.4.0-py3-none-any.whl (236 kB)
Collecting rich>=10.0.0
  Using cached rich-13.0.0-py3-none-any.whl (238 kB)
Collecting rq>=1.2
  Using cached rq-1.11.1-py2.py3-none-any.whl (70 kB)
Collecting pytz
  Using cached pytz-2022.7-py2.py3-none-any.whl (499 kB)
Collecting ruamel.yaml>=0.16.13
  Using cached ruamel.yaml-0.17.21-py3-none-any.whl (109 kB)
Collecting inflection>=0.3.1
  Using cached inflection-0.5.1-py2.py3-none-any.whl (9.5 kB)
Collecting uritemplate>=3.0.0
  Using cached uritemplate-4.1.1-py2.py3-none-any.whl (10 kB)
Collecting coreapi>=2.3.3
  Using cached coreapi-2.3.3-py2.py3-none-any.whl (25 kB)
Collecting packaging>=21.0
  Using cached packaging-22.0-py3-none-any.whl (42 kB)
Collecting coreschema>=0.0.4
  Using cached coreschema-0.0.4-py3-none-any.whl
Collecting swagger-spec-validator>=2.1.0
  Using cached swagger_spec_validator-3.0.3-py2.py3-none-any.whl (27 kB)
Collecting promise>=2.1
  Using cached promise-2.3-py3-none-any.whl
Collecting text-unidecode
  Using cached text_unidecode-1.3-py2.py3-none-any.whl (78 kB)
Collecting graphql-core<4,>=3.1.0
  Using cached graphql_core-3.2.3-py3-none-any.whl (202 kB)
Collecting graphene<4,>=3.0
  Using cached graphene-3.2.1-py2.py3-none-any.whl (125 kB)
Collecting graphql-relay<4,>=3.1.1
  Using cached graphql_relay-3.2.0-py3-none-any.whl (16 kB)
Requirement already satisfied: setuptools>=3.0 in ./venv/lib/python3.9/site-packages (from gunicorn==20.1.0->-r requirements.txt (line 19)) (58.1.0)
Collecting MarkupSafe>=2.0
  Using cached MarkupSafe-2.1.1-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (25 kB)
Collecting importlib-metadata>=4.4
  Using cached importlib_metadata-5.2.0-py3-none-any.whl (21 kB)
Collecting mkdocs-material-extensions>=1.1
  Using cached mkdocs_material_extensions-1.1.1-py3-none-any.whl (7.9 kB)
Collecting pygments>=2.12
  Using cached Pygments-2.13.0-py3-none-any.whl (1.1 MB)
Collecting pymdown-extensions>=9.4
  Using cached pymdown_extensions-9.9-py3-none-any.whl (219 kB)
Collecting mkdocs>=1.4.0
  Using cached mkdocs-1.4.2-py3-none-any.whl (3.7 MB)
Collecting requests>=2.26
  Using cached requests-2.28.1-py3-none-any.whl (62 kB)
Collecting mkdocs-autorefs>=0.3.1
  Using cached mkdocs_autorefs-0.4.1-py3-none-any.whl (9.8 kB)
Collecting mkdocstrings-python-legacy>=0.2.1
  Using cached mkdocstrings_python_legacy-0.2.3-py3-none-any.whl (26 kB)
Collecting urllib3>=1.26.11
  Using cached urllib3-1.26.13-py2.py3-none-any.whl (140 kB)
Collecting certifi
  Using cached certifi-2022.12.7-py3-none-any.whl (155 kB)
Collecting PyJWT>=2.0.0
  Using cached PyJWT-2.6.0-py3-none-any.whl (20 kB)
Collecting requests-oauthlib>=0.6.1
  Using cached requests_oauthlib-1.3.1-py2.py3-none-any.whl (23 kB)
Collecting defusedxml>=0.5.0rc1
  Using cached defusedxml-0.7.1-py2.py3-none-any.whl (25 kB)
Collecting oauthlib>=1.0.3
  Using cached oauthlib-3.2.2-py3-none-any.whl (151 kB)
Collecting cryptography>=1.4
  Using cached cryptography-38.0.4-cp36-abi3-manylinux_2_24_x86_64.whl (4.0 MB)
Collecting python3-openid>=3.0.10
  Using cached python3_openid-3.2.0-py3-none-any.whl (133 kB)
Collecting python-jose>=3.0.0
  Using cached python_jose-3.3.0-py2.py3-none-any.whl (33 kB)
Collecting pyrsistent>=0.14.0
  Using cached pyrsistent-0.19.3-py3-none-any.whl (57 kB)
Collecting attrs>=17.4.0
  Using cached attrs-22.2.0-py3-none-any.whl (60 kB)
Collecting itypes
  Using cached itypes-1.2.0-py2.py3-none-any.whl (4.8 kB)
Collecting cffi>=1.12
  Using cached cffi-1.15.1-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (441 kB)
Collecting aniso8601<10,>=8
  Using cached aniso8601-9.0.1-py2.py3-none-any.whl (52 kB)
Collecting zipp>=0.5
  Using cached zipp-3.11.0-py3-none-any.whl (6.6 kB)
Collecting ghp-import>=1.0
  Using cached ghp_import-2.1.0-py3-none-any.whl (11 kB)
Collecting mergedeep>=1.3.4
  Using cached mergedeep-1.3.4-py3-none-any.whl (6.4 kB)
Collecting click>=7.0
  Using cached click-8.1.3-py3-none-any.whl (96 kB)
Collecting pyyaml-env-tag>=0.1
  Using cached pyyaml_env_tag-0.1-py3-none-any.whl (3.9 kB)
Collecting watchdog>=2.0
  Using cached watchdog-2.2.0-py3-none-manylinux2014_x86_64.whl (78 kB)
Collecting pytkdocs>=0.14
  Using cached pytkdocs-0.16.1-py3-none-any.whl (37 kB)
Collecting pyasn1
  Using cached pyasn1-0.4.8-py2.py3-none-any.whl (77 kB)
Collecting ecdsa!=0.15
  Using cached ecdsa-0.18.0-py2.py3-none-any.whl (142 kB)
Collecting rsa
  Using cached rsa-4.9-py3-none-any.whl (34 kB)
Collecting async-timeout>=4.0.2
  Using cached async_timeout-4.0.2-py3-none-any.whl (5.8 kB)
Collecting charset-normalizer<3,>=2
  Using cached charset_normalizer-2.1.1-py3-none-any.whl (39 kB)
Collecting idna<4,>=2.5
  Using cached idna-3.4-py3-none-any.whl (61 kB)
Collecting commonmark<0.10.0,>=0.9.0
  Using cached commonmark-0.9.1-py2.py3-none-any.whl (51 kB)
Collecting ruamel.yaml.clib>=0.2.6
  Using cached ruamel.yaml.clib-0.2.7-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.manylinux_2_24_x86_64.whl (519 kB)
Collecting typing-extensions
  Using cached typing_extensions-4.4.0-py3-none-any.whl (26 kB)
Collecting pycparser
  Using cached pycparser-2.21-py2.py3-none-any.whl (118 kB)
Collecting python-dateutil>=2.8.1
  Using cached python_dateutil-2.8.2-py2.py3-none-any.whl (247 kB)
Installing collected packages: webencodings, text-unidecode, pytz, pyasn1, netaddr, itypes, commonmark, aniso8601, zipp, watchdog, urllib3, uritemplate, tzdata, typing-extensions, tablib, svgwrite, sqlparse, six, ruamel.yaml.clib, rsa, PyYAML, pytkdocs, pyrsistent, PyJWT, pygments, pycparser, psycopg2-binary, prometheus-client, Pillow, packaging, oauthlib, mkdocs-material-extensions, mergedeep, MarkupSafe, inflection, idna, gunicorn, graphql-core, django-timezone-field, defusedxml, click, charset-normalizer, certifi, attrs, async-timeout, asgiref, sentry-sdk, ruamel.yaml, rich, requests, redis, pyyaml-env-tag, python3-openid, python-dateutil, promise, jsonschema, Jinja2, importlib-metadata, graphql-relay, ecdsa, django-prometheus, django-pglocks, Django, cffi, bleach, swagger-spec-validator, rq, requests-oauthlib, python-jose, Markdown, graphene, ghp-import, djangorestframework, django-taggit, django-tables2, django-rich, django-redis, django-js-asset, django-filter, django-debug-toolbar, django-cors-headers, cryptography, coreschema, social-auth-core, pymdown-extensions, mkdocs, graphene-django, django-rq, django-mptt, coreapi, social-auth-app-django, mkdocs-material, mkdocs-autorefs, drf-yasg, django-graphiql-debug-toolbar, mkdocstrings, mkdocstrings-python-legacy
Successfully installed Django-4.1.4 Jinja2-3.1.2 Markdown-3.3.7 MarkupSafe-2.1.1 Pillow-9.3.0 PyJWT-2.6.0 PyYAML-6.0 aniso8601-9.0.1 asgiref-3.6.0 async-timeout-4.0.2 attrs-22.2.0 bleach-5.0.1 certifi-2022.12.7 cffi-1.15.1 charset-normalizer-2.1.1 click-8.1.3 commonmark-0.9.1 coreapi-2.3.3 coreschema-0.0.4 cryptography-38.0.4 defusedxml-0.7.1 django-cors-headers-3.13.0 django-debug-toolbar-3.8.1 django-filter-22.1 django-graphiql-debug-toolbar-0.2.0 django-js-asset-2.0.0 django-mptt-0.14.0 django-pglocks-1.0.4 django-prometheus-2.2.0 django-redis-5.2.0 django-rich-1.4.0 django-rq-2.6.0 django-tables2-2.4.1 django-taggit-3.1.0 django-timezone-field-5.0 djangorestframework-3.14.0 drf-yasg-1.21.4 ecdsa-0.18.0 ghp-import-2.1.0 graphene-3.2.1 graphene-django-3.0.0 graphql-core-3.2.3 graphql-relay-3.2.0 gunicorn-20.1.0 idna-3.4 importlib-metadata-5.2.0 inflection-0.5.1 itypes-1.2.0 jsonschema-3.2.0 mergedeep-1.3.4 mkdocs-1.4.2 mkdocs-autorefs-0.4.1 mkdocs-material-8.5.11 mkdocs-material-extensions-1.1.1 mkdocstrings-0.19.1 mkdocstrings-python-legacy-0.2.3 netaddr-0.8.0 oauthlib-3.2.2 packaging-22.0 prometheus-client-0.15.0 promise-2.3 psycopg2-binary-2.9.5 pyasn1-0.4.8 pycparser-2.21 pygments-2.13.0 pymdown-extensions-9.9 pyrsistent-0.19.3 python-dateutil-2.8.2 python-jose-3.3.0 python3-openid-3.2.0 pytkdocs-0.16.1 pytz-2022.7 pyyaml-env-tag-0.1 redis-4.4.0 requests-2.28.1 requests-oauthlib-1.3.1 rich-13.0.0 rq-1.11.1 rsa-4.9 ruamel.yaml-0.17.21 ruamel.yaml.clib-0.2.7 sentry-sdk-1.11.1 six-1.16.0 social-auth-app-django-5.0.0 social-auth-core-4.3.0 sqlparse-0.4.3 svgwrite-1.4.3 swagger-spec-validator-3.0.3 tablib-3.3.0 text-unidecode-1.3 typing-extensions-4.4.0 tzdata-2022.7 uritemplate-4.1.1 urllib3-1.26.13 watchdog-2.2.0 webencodings-0.5.1 zipp-3.11.0
Installing local dependencies (pip install -r local_requirements.txt)...
Requirement already satisfied: setuptools>=36 in ./venv/lib/python3.9/site-packages (from -r local_requirements.txt (line 1)) (58.1.0)
Collecting napalm
  Using cached napalm-4.0.0-py2.py3-none-any.whl (266 kB)
Collecting napalm-ros
  Using cached napalm_ros-1.2.2-py3-none-any.whl
Collecting netbox-secretstore
  Using cached netbox_secretstore-1.4.1-py3-none-any.whl (66 kB)
Collecting netbox-qrcode
  Using cached netbox_qrcode-0.0.8-py3-none-any.whl (1.7 MB)
Collecting django-auth-ldap
  Using cached django_auth_ldap-4.1.0-py3-none-any.whl (20 kB)
Collecting netmiko>=4.0.0
  Using cached netmiko-4.1.2-py3-none-any.whl (196 kB)
Collecting pyeapi>=0.8.2
  Using cached pyeapi-0.8.4-py3-none-any.whl
Collecting ttp-templates
  Using cached ttp_templates-0.3.2-py3-none-any.whl (71 kB)
Collecting junos-eznc>=2.6.3
  Using cached junos_eznc-2.6.6-py2.py3-none-any.whl (197 kB)
Requirement already satisfied: jinja2 in ./venv/lib/python3.9/site-packages (from napalm->-r local_requirements.txt (line 2)) (3.1.2)
Collecting paramiko>=2.6.0
  Using cached paramiko-2.12.0-py2.py3-none-any.whl (213 kB)
Collecting future
  Using cached future-0.18.2-py3-none-any.whl
Collecting textfsm<=1.1.2
  Using cached textfsm-1.1.2-py2.py3-none-any.whl (44 kB)
Requirement already satisfied: typing-extensions>=4.3.0 in ./venv/lib/python3.9/site-packages (from napalm->-r local_requirements.txt (line 2)) (4.4.0)
Collecting netutils>=1.0.0
  Using cached netutils-1.3.0-py3-none-any.whl (479 kB)
Requirement already satisfied: netaddr in ./venv/lib/python3.9/site-packages (from napalm->-r local_requirements.txt (line 2)) (0.8.0)
Requirement already satisfied: pyYAML in ./venv/lib/python3.9/site-packages (from napalm->-r local_requirements.txt (line 2)) (6.0)
Collecting ttp
  Using cached ttp-0.9.2-py2.py3-none-any.whl (85 kB)
Collecting lxml>=4.3.0
  Using cached lxml-4.9.2-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.manylinux_2_24_x86_64.whl (7.1 MB)
Requirement already satisfied: cffi>=1.11.3 in ./venv/lib/python3.9/site-packages (from napalm->-r local_requirements.txt (line 2)) (1.15.1)
Collecting scp
  Using cached scp-0.14.4-py2.py3-none-any.whl (8.6 kB)
Collecting ncclient
  Using cached ncclient-0.6.13-py2.py3-none-any.whl
Requirement already satisfied: requests>=2.7.0 in ./venv/lib/python3.9/site-packages (from napalm->-r local_requirements.txt (line 2)) (2.28.1)
Collecting librouteros==3.*
  Using cached librouteros-3.2.1-py3-none-any.whl
Collecting pycryptodome
  Using cached pycryptodome-3.16.0-cp35-abi3-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_12_x86_64.manylinux2010_x86_64.whl (2.3 MB)
Collecting importlib
  Using cached importlib-1.0.4-py3-none-any.whl
Collecting qrcode
  Using cached qrcode-7.3.1-py3-none-any.whl
Requirement already satisfied: Pillow in ./venv/lib/python3.9/site-packages (from netbox-qrcode->-r local_requirements.txt (line 7)) (9.3.0)
Collecting python-ldap>=3.1
  Using cached python_ldap-3.4.3-cp39-cp39-linux_x86_64.whl
Requirement already satisfied: Django>=2.2 in ./venv/lib/python3.9/site-packages (from django-auth-ldap->-r local_requirements.txt (line 8)) (4.1.4)
Requirement already satisfied: pycparser in ./venv/lib/python3.9/site-packages (from cffi>=1.11.3->napalm->-r local_requirements.txt (line 2)) (2.21)
Requirement already satisfied: asgiref<4,>=3.5.2 in ./venv/lib/python3.9/site-packages (from Django>=2.2->django-auth-ldap->-r local_requirements.txt (line 8)) (3.6.0)
Requirement already satisfied: sqlparse>=0.2.2 in ./venv/lib/python3.9/site-packages (from Django>=2.2->django-auth-ldap->-r local_requirements.txt (line 8)) (0.4.3)
Requirement already satisfied: six in ./venv/lib/python3.9/site-packages (from junos-eznc>=2.6.3->napalm->-r local_requirements.txt (line 2)) (1.16.0)
Collecting transitions
  Using cached transitions-0.9.0-py2.py3-none-any.whl (97 kB)
Collecting pyserial
  Using cached pyserial-3.5-py2.py3-none-any.whl (90 kB)
Collecting pyparsing
  Using cached pyparsing-3.0.9-py3-none-any.whl (98 kB)
Collecting yamlordereddictloader
  Using cached yamlordereddictloader-0.4.0-py3-none-any.whl
Requirement already satisfied: MarkupSafe>=2.0 in ./venv/lib/python3.9/site-packages (from jinja2->napalm->-r local_requirements.txt (line 2)) (2.1.1)
Collecting ntc-templates>=2.0.0
  Using cached ntc_templates-3.2.0-py3-none-any.whl (354 kB)
Collecting tenacity
  Using cached tenacity-8.1.0-py3-none-any.whl (23 kB)
Collecting pynacl>=1.0.1
  Using cached PyNaCl-1.5.0-cp36-abi3-manylinux_2_17_x86_64.manylinux2014_x86_64.manylinux_2_24_x86_64.whl (856 kB)
Collecting bcrypt>=3.1.3
  Using cached bcrypt-4.0.1-cp36-abi3-manylinux_2_24_x86_64.whl (593 kB)
Requirement already satisfied: cryptography>=2.5 in ./venv/lib/python3.9/site-packages (from paramiko>=2.6.0->napalm->-r local_requirements.txt (line 2)) (38.0.4)
Collecting pyasn1-modules>=0.1.5
  Using cached pyasn1_modules-0.2.8-py2.py3-none-any.whl (155 kB)
Requirement already satisfied: pyasn1>=0.3.7 in ./venv/lib/python3.9/site-packages (from python-ldap>=3.1->django-auth-ldap->-r local_requirements.txt (line 8)) (0.4.8)
Requirement already satisfied: charset-normalizer<3,>=2 in ./venv/lib/python3.9/site-packages (from requests>=2.7.0->napalm->-r local_requirements.txt (line 2)) (2.1.1)
Requirement already satisfied: idna<4,>=2.5 in ./venv/lib/python3.9/site-packages (from requests>=2.7.0->napalm->-r local_requirements.txt (line 2)) (3.4)
Requirement already satisfied: certifi>=2017.4.17 in ./venv/lib/python3.9/site-packages (from requests>=2.7.0->napalm->-r local_requirements.txt (line 2)) (2022.12.7)
Requirement already satisfied: urllib3<1.27,>=1.21.1 in ./venv/lib/python3.9/site-packages (from requests>=2.7.0->napalm->-r local_requirements.txt (line 2)) (1.26.13)
Installing collected packages: pyserial, librouteros, importlib, yamlordereddictloader, ttp, transitions, tenacity, qrcode, pyparsing, pyeapi, pycryptodome, pyasn1-modules, netutils, lxml, future, bcrypt, ttp-templates, textfsm, python-ldap, pynacl, netbox-secretstore, netbox-qrcode, paramiko, ntc-templates, django-auth-ldap, scp, ncclient, netmiko, junos-eznc, napalm, napalm-ros
Successfully installed bcrypt-4.0.1 django-auth-ldap-4.1.0 future-0.18.2 importlib-1.0.4 junos-eznc-2.6.6 librouteros-3.2.1 lxml-4.9.2 napalm-4.0.0 napalm-ros-1.2.2 ncclient-0.6.13 netbox-qrcode-0.0.8 netbox-secretstore-1.4.1 netmiko-4.1.2 netutils-1.3.0 ntc-templates-3.2.0 paramiko-2.12.0 pyasn1-modules-0.2.8 pycryptodome-3.16.0 pyeapi-0.8.4 pynacl-1.5.0 pyparsing-3.0.9 pyserial-3.5 python-ldap-3.4.3 qrcode-7.3.1 scp-0.14.4 tenacity-8.1.0 textfsm-1.1.2 transitions-0.9.0 ttp-0.9.2 ttp-templates-0.3.2 yamlordereddictloader-0.4.0
Applying database migrations (python3 netbox/manage.py migrate)...
Traceback (most recent call last):
  File "/opt/netbox/netbox/manage.py", line 10, in <module>
    execute_from_command_line(sys.argv)
  File "/opt/netbox/venv/lib/python3.9/site-packages/django/core/management/__init__.py", line 446, in execute_from_command_line
    utility.execute()
  File "/opt/netbox/venv/lib/python3.9/site-packages/django/core/management/__init__.py", line 440, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/opt/netbox/venv/lib/python3.9/site-packages/django/core/management/base.py", line 402, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/opt/netbox/venv/lib/python3.9/site-packages/django/core/management/base.py", line 448, in execute
    output = self.handle(*args, **options)
  File "/opt/netbox/venv/lib/python3.9/site-packages/django/core/management/base.py", line 96, in wrapped
    res = handle_func(*args, **kwargs)
  File "/opt/netbox/venv/lib/python3.9/site-packages/django/core/management/commands/migrate.py", line 114, in handle
    executor = MigrationExecutor(connection, self.migration_progress_callback)
  File "/opt/netbox/venv/lib/python3.9/site-packages/django/db/migrations/executor.py", line 18, in __init__
    self.loader = MigrationLoader(self.connection)
  File "/opt/netbox/venv/lib/python3.9/site-packages/django/db/migrations/loader.py", line 58, in __init__
    self.build_graph()
  File "/opt/netbox/venv/lib/python3.9/site-packages/django/db/migrations/loader.py", line 235, in build_graph
    self.applied_migrations = recorder.applied_migrations()
  File "/opt/netbox/venv/lib/python3.9/site-packages/django/db/migrations/recorder.py", line 81, in applied_migrations
    if self.has_table():
  File "/opt/netbox/venv/lib/python3.9/site-packages/django/db/migrations/recorder.py", line 57, in has_table
    with self.connection.cursor() as cursor:
  File "/opt/netbox/venv/lib/python3.9/site-packages/django/utils/asyncio.py", line 26, in inner
    return func(*args, **kwargs)
  File "/opt/netbox/venv/lib/python3.9/site-packages/django/db/backends/base/base.py", line 323, in cursor
    return self._cursor()
  File "/opt/netbox/venv/lib/python3.9/site-packages/django/db/backends/base/base.py", line 299, in _cursor
    self.ensure_connection()
  File "/opt/netbox/venv/lib/python3.9/site-packages/django/utils/asyncio.py", line 26, in inner
    return func(*args, **kwargs)
  File "/opt/netbox/venv/lib/python3.9/site-packages/django/db/backends/base/base.py", line 282, in ensure_connection
    self.connect()
  File "/opt/netbox/venv/lib/python3.9/site-packages/django/utils/asyncio.py", line 26, in inner
    return func(*args, **kwargs)
  File "/opt/netbox/venv/lib/python3.9/site-packages/django/db/backends/base/base.py", line 265, in connect
    self.init_connection_state()
  File "/opt/netbox/venv/lib/python3.9/site-packages/django/db/backends/postgresql/base.py", line 251, in init_connection_state
    super().init_connection_state()
  File "/opt/netbox/venv/lib/python3.9/site-packages/django/db/backends/base/base.py", line 232, in init_connection_state
    self.check_database_version_supported()
  File "/opt/netbox/venv/lib/python3.9/site-packages/django/db/backends/base/base.py", line 207, in check_database_version_supported
    raise NotSupportedError(
django.db.utils.NotSupportedError: PostgreSQL 11 or later is required (found 10.22).
`

**Comment by Delta1977** [comment]
Hi, from releasenotes:

WARNING: NetBox v3.4 requires PostgreSQL 11 or later.

**Reply by Abulhallaj** [reply]
How to painlessly upgrade Postgre?
